lxd/locking/prune_lock.go

```diff
@@ -0,0 +1,45 @@
+package locking
+
+import "sync"
+
+// PruneLock tracks a pruning items
+type PruneLock struct {
```

Comment:
This type is very specific to snapshot pruning and yet this package `locking` is a general package, so not really well suited here. If it was renamed to `MapLock` or similar and the key type changed to `interface{}` then it would be more suitable. But it would then basically offer the same thing as https://pkg.go.dev/sync#Map in which case we should probably just use that.

Reply:
I've removed `PruneLock` struct and used `sync.Map`.